### ISSUE TITLE
Drop computation of package/file tangles when dropping support for SQ's < 5.0

### ISSUE THREAD

**Issue by wenns** [SonarOpenCommunity/sonar-cxx#364]
The DSM-Matrix calculations are moved into the core with 5.0. We will have to rework our package/file tangles implementation as soon as we drop the support for version prior to 5.0.

For details, see:
http://sonarqube.15.x6.nabble.com/sonar-dev-Multi-language-support-for-package-cycles-tt5031235.html

**Comment by guwirth** [comment]
http://jira.codehaus.org/browse/SONAR-5672.
http://jira.codehaus.org/browse/SONARJAVA-664

As soon as your plugin will no more support version prior to 5.0 you could safely drop computation of design metrics since it will be done by the platform based on declared file dependencies.

**Comment by guwirth** [comment]
Can not be solved with 4.5.2 API

**Comment by guwirth** [comment]
@all someone familiar with this? Do we have to remove this completely or partly or have to use a new API?

**Comment by Bertk** [comment]
I think DSM shall be removed completely from C++ plug-in. This feature was anyway not working for a multi-module solution and as far as I know only used with java plug-in.

**Comment by jmecosta** [comment]
Perhaps we have been approaching the problem incorrectly, imo detecting cycles between modules is the only way this makes sense. So maybe we should try to improve this to only report issues in such cases. We dont need a fancy graph in dashboard.  Having cycles detected as issues is already a quite powerful feature.

**Comment by Typz** [comment]
As far as I know, with latest versions we should only provide the file dependencies, and the rest will be computed automatically. This means the following metrics should not be computed/reported:
    CoreMetrics.DEPENDENCY_MATRIX,
    CoreMetrics.PACKAGE_CYCLES,
    CoreMetrics.PACKAGE_EDGES_WEIGHT,
    CoreMetrics.PACKAGE_FEEDBACK_EDGES,
    CoreMetrics.PACKAGE_TANGLE_INDEX,
    CoreMetrics.PACKAGE_TANGLES,
    CoreMetrics.FILE_CYCLES,
    CoreMetrics.FILE_EDGES_WEIGHT,
    CoreMetrics.FILE_FEEDBACK_EDGES,
    CoreMetrics.FILE_TANGLE_INDEX,
    CoreMetrics.FILE_TANGLES

But we should still provide all the dependencies, as well as report the issues.

**Comment by Typz** [comment]
After quick look at the file, I think that would be removing lines 160-177 (DependencyAnalyzer.save()) and lines 188-203 (DependencyAnalyzer.saveDirectory()).

Note that this has not yet been removed from Java plugin either.

**Comment by guwirth** [comment]
Hello @Typz, thanks for looking to this.  Our next release will focus on SQ 5.6 LTS support. As far as I know most other plugins are still using the SQ 4.5 APIs. We did a first try to update to latest API but we're running into problems with Squid dependencies. After SQ 5.5 is released I hope all others are switching to 5.6 API now. So we can have a look how others are doing it. I already started to mark deprecated APIs - there are quit a lot. So any help to keep this plugin up to date is welcome...
